### PR TITLE
remove IHttpContextAccessor

### LINKED ISSUE #32: remove IHttpContextAccessor and use HttpContext as param
https://twitter.com/davidfowl/status/1563935285754593281

### PR DESCRIPTION
**Pull request by BenjaminAbt**
fix #32

### CODE REVIEW

File: src/MyCSharp.HttpUserAgentParser/HttpContextExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+/// <summary>
+/// Static extensions for <see cref="HttpContext"/>
+/// </summary>
+public static class HttpContextExtensions
+{
+    /// <summary>
+    /// Returns the User-Agent header value
+    /// </summary>
+    public static string? GetUserAgentString(this HttpContext httpContext) =>
+         httpContext.Request?.Headers["User-Agent"].ToString()!;
```
Comment (gfoidl) [inline src/MyCSharp.HttpUserAgentParser/HttpContextExtensions.cs:16]:
```suggestion
    public static string? GetUserAgentString(this HttpContext httpContext)
    {
        if (httpContext.Request.Headers.TryGetValue("User-Agent", out StringValues value))
        {
            return value;
        }

        return null;
    }
```
Or is it safe to assume that the UA is always present?

File: version.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
     "$schema": "https://raw.githubusercontent.com/dotnet/Nerdbank.GitVersioning/master/src/NerdBank.GitVersioning/version.schema.json",
-    "version": "1.1",
+    "version": "2.0",
```
Comment (gfoidl) [inline version.json:3]: Is this the update for the major version due the breaking change (comment above)?
If so, perfect.

File: .editorconfig [hunk 3]
```diff
@@ -19,7 +19,7 @@ trim_trailing_whitespace = true
 # C# files
 [*.cs]
 
-file_header_template = Copyright © myCSharp 2020-2021, all rights reserved
+file_header_template = Copyright © myCSharp 2020-2022, all rights reserved
```
Comment (gfoidl) [inline .editorconfig:22]: Do we need to specify the year at all?
So we have to update next year again...
Is 
```suggestion
file_header_template = Copyright © myCSharp, all rights reserved
```
sufficient? I see quite a lot other repos that don't specify the year too.

File: src/MyCSharp.HttpUserAgentParser.AspNetCore/HttpUserAgentParserAccessor.cs [hunk 4]
```diff
@@ -13,41 +13,47 @@ public interface IHttpUserAgentParserAccessor
         /// <summary>
         /// User agent value
         /// </summary>
-        string HttpContextUserAgent { get; }
+        string? GetHttpContextUserAgent(HttpContext httpContext);
```
[the remark this answers is not in the document]
Comment (BenjaminAbt) [reply src/MyCSharp.HttpUserAgentParser.AspNetCore/HttpUserAgentParserAccessor.cs:16]: Already part of this PR. New version generated:

> Successfully created package '/home/runner/work/HttpUserAgentParser/HttpUserAgentParser/src/MyCSharp.HttpUserAgentParser.AspNetCore/bin/Release/MyCSharp.HttpUserAgentParser.AspNetCore.2.0.3-g19e29e53bc.nupkg'.

File: src/MyCSharp.HttpUserAgentParser.MemoryCache/HttpUserAgentParserMemoryCachedProvider.cs [hunk 5]
```diff
@@ -15,10 +15,11 @@ public class HttpUserAgentParserMemoryCachedProvider : IHttpUserAgentParserProvi
         /// <summary>
         /// Creates a new instance of <see cref="HttpUserAgentParserMemoryCachedProvider"/>.
         /// </summary>
+        /// <param name="memoryCache">The memory cache instance to use</param>
         /// <param name="options">The options used to set expiration and size limit</param>
-        public HttpUserAgentParserMemoryCachedProvider(HttpUserAgentParserMemoryCachedProviderOptions options)
+        public HttpUserAgentParserMemoryCachedProvider(IMemoryCache memoryCache, HttpUserAgentParserMemoryCachedProviderOptions options)
         {
-            _memoryCache = new Microsoft.Extensions.Caching.Memory.MemoryCache(options.CacheOptions);
+            _memoryCache = memoryCache;
```
[the remark this answers is not in the document]
Comment (BenjaminAbt) [reply src/MyCSharp.HttpUserAgentParser.MemoryCache/HttpUserAgentParserMemoryCachedProvider.cs:22]: I guess this automatically happaned due to "Code Cleanup", no idea why

File: version.json (hunk 2 above)
Comment (BenjaminAbt) [reply version.json:3]: Yes

File: src/MyCSharp.HttpUserAgentParser.MemoryCache/HttpUserAgentParserMemoryCachedProvider.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (BenjaminAbt) [reply src/MyCSharp.HttpUserAgentParser.MemoryCache/HttpUserAgentParserMemoryCachedProvider.cs:22]: I just have pull'ed main.
I did not change that code. Weird.

**Review by gfoidl — APPROVED** [review]

**Pull request merged**